bin/agitator

```diff
@@ -30,6 +30,144 @@ Possible commands:
 EOF
 }
 
+function run_agitators() {
+  unset APP_HOSTS
+  export APP_HOSTS=$MANAGER_HOSTS
+  echo "Starting manager agitation as ${1} for ${APP_HOSTS}"
+  start_app_agitator "manager" "${AGTR_MASTER_KILL_SLEEP_TIME}" "${AGTR_MASTER_RESTART_SLEEP_TIME}" 1 1 > "${manager_log}" 2>&1 &
+  unset APP_HOSTS
+  export APP_HOSTS=$TSERVER_HOSTS
+  echo "Starting tserver agitation as ${1} for ${APP_HOSTS}"
+  start_app_agitator "tserver" "${AGTR_TSERVER_KILL_SLEEP_TIME}" "${AGTR_TSERVER_RESTART_SLEEP_TIME}" "${AGTR_TSERVER_MIN_KILL}" "${AGTR_TSERVER_MAX_KILL}" > "${tserver_log}" 2>&1 &
+}
+
+# Starts a app specific agitator
+# usage: start_app_agitator app_name kill_sleep_time restart_sleep_time min_kill max_kill
+# Requires that a list of hosts to be set in $APP_HOSTS
+function start_app_agitator() {
+  app_name=$1
+  kill_sleep_time=$2
+  restart_sleep_time=$3
+  min_kill=$4
+  max_kill=$5
```

Comment:
```suggestion
# Requires that a list of hosts to be set in $APP_HOSTS
function start_app_agitator() {
  local app_name=$1; shift
  local agtr_user=$1; shift
  local kill_sleep_time=$1; shift
  local restart_sleep_time=$1; shift
  local min_kill=$1; shift
  local max_kill=$1; shift
  local APP_HOSTS=("$@") # can keep this as an array
  echo "Starting $app_name agitation as $agtr_user for ${APP_HOSTS[*]}"
```

Reply:
What are the benefits of using shift 6 times? Personally I think this is a lot less user friendly then using the numbers for the parameters. If someone comes and looks at the script to figure out the params they have to do counting and also the shifting makes the ordering of setting the variables important.

Reply:
I see why you are shifting; so you can set the last one as an array. I think I am going to create another function to set APP_HOSTS so this is more readable.